bin/yunopaste

```diff
@@ -1,77 +1,64 @@
-#!/bin/bash
-
-set -e
-set -u
-
-PASTE_URL="https://paste.yunohost.org"
-
-_die() {
-  printf "Error: %s\n" "$*"
-  exit 1
-}
-
-check_dependencies() {
-  curl -V > /dev/null 2>&1 || _die "This script requires curl."
-}
-
-paste_data() {
-  json=$(curl -X POST -s -d "$1" "${PASTE_URL}/documents")
-  [[ -z "$json" ]] && _die "Unable to post the data to the server."
-
-  key=$(echo "$json" \
-    | python3 -c 'import json,sys;o=json.load(sys.stdin);print(o["key"])' \
-        2>/dev/null)
-  [[ -z "$key" ]] && _die "Unable to parse the server response."
-
-  echo "${PASTE_URL}/${key}"
-}
-
-usage() {
-  printf "Usage: ${0} [OPTION]...
-
-Read from input stream and paste the data to the YunoHost
-Haste server.
-
-For example, to paste the output of the YunoHost diagnosis, you
-can simply execute the following:
-  yunohost diagnosis show | ${0}
-
-It will return the URL where you can access the pasted data.
-
-Options:
-  -h, --help   show this help message and exit
-"
-}
-
-main() {
-  # parse options
-  while (( ${#} )); do
-    case "${1}" in
-      --help|-h)
-        usage
-        exit 0
-        ;;
-      *)
-        echo "Unknown parameter detected: ${1}" >&2
-        echo >&2
-        usage >&2
-        exit 1
-        ;;
-    esac
-
-    shift 1
-  done
-
-  # check input stream
-  read -t 0 || {
-    echo -e "Invalid usage: No input is provided.\n" >&2
-    usage
-    exit 1
-  }
-
-  paste_data "$(cat)"
-}
-
-check_dependencies
-
-main "${@}"
+#!/usr/bin/env python3
+
+import sys
+import requests
+import json
+import re
+
+SERVER_URL = "https://paste.yunohost.org"
+TIMEOUT = 3
+
+def create_snippet(data):
+    try:
+        url = SERVER_URL + "/documents"
+        response = requests.post(url, data=data.encode('utf-8'), timeout=TIMEOUT)
+        response.raise_for_status()
+        dockey = json.loads(response.text)['key']
+        return SERVER_URL + "/" + dockey
+    except requests.exceptions.RequestException as e:
+        print("\033[31mError: {}\033[0m".format(e))
+        sys.exit(1)
+
+def anonymize_output(output):
+    # Replace IPv4 addresses
+    output = re.sub(r"\b(?:\d{1,3}\.){3}\d{1,3}\b", "[IPv4]", output)
+    # Replace IPv6 addresses
+    output = re.sub(r"\b(?:[A-Fa-f0-9]{1,4}:){7}[A-Fa-f0-9]{1,4}\b", "[IPv6]", output)
+    # Replace domain names
+    output = re.sub(r"\b(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}\b", "[DOMAIN]", output)
```

Comment:
This regex is a bit too brutal ... It finds anything looking like `foo.bar` as a domain, such as for example `some_app.conf` or `manifest.json`

Resulting in confusing lines such as : `chown tiki:tiki /etc/nginx/conf.d/[DOMAIN].d/[DOMAIN]`

or : `manifest=/var/cache/yunohost/app_tmp_work_dirs/app_6k1a4ef4/[DOMAIN]`

To a lesser extend, also the apt install stuff: `Get:1 http://[DOMAIN]/debian stable/main amd64 libgif7 amd64 5.1.9-2 [45.1 kB]`

Personally I would tend to only replace domains registered in YunoHost ... the list can be obtained in various ways, but for example finding all occurences of folders named `/etc/nginx/conf.d/DOMAIN.d/`